callable/src/main/java/io/grpc/callable/Callable.java

```diff
@@ -0,0 +1,423 @@
+  //             much that we want to rename this into ClientCallable or such.
+
+  // Subclass Contract
+  // =================
+
+  /**
+   * Creates a new GRPC call from this callable. A channel may or may not be provided.
+   * If a channel is not provided, the callable must be bound to a channel.
+   */
+  public abstract ClientCall<RequestT, ResponseT> newCall(@Nullable Channel channel);
+
+  /**
+   * Return a descriptor for this callable, or null if none available.
+   */
+  @Nullable public CallableDescriptor<RequestT, ResponseT> getDescriptor() {
+    return null;
+  }
+
+  /**
+   * Gets the channel bound to this callable, or null, if none is bound to it.
+   */
+  @Nullable public Channel getBoundChannel() {
+    return null;
+  }
+
+  // Binding Callables
+  // =================
+
+  /**
+   * Returns a callable which is bound to the given channel. Operations on the result can
+   * omit the channel. If a channel is provided anyway, it overrides the bound channel.
+   */
+  public Callable<RequestT, ResponseT> bind(final Channel boundChannel) {
+    return new Callable<RequestT, ResponseT>() {
+      @Override
+      public ClientCall<RequestT, ResponseT> newCall(@Nullable Channel channel) {
+        if (channel == null) {
+          // If the caller does not provide a channel, we use the bound one.
+          channel = boundChannel;
+        }
+        return Callable.this.newCall(channel);
+      }
+
+      @Override
+      @Nullable
+      public CallableDescriptor<RequestT, ResponseT> getDescriptor() {
+        return Callable.this.getDescriptor();
+      }
+
+      @Override
+      @Nullable
+      public Channel getBoundChannel() {
+        return boundChannel;
+      }
+    };
+  }
+
+  // Running Callables
+  // =================
+
+  /**
+   * Convenience method to run a unary callable synchronously. If no channel is provided,
+   * the callable must be bound to one.
+   */
+  public ResponseT call(@Nullable Channel channel, RequestT request) {
+    return ClientCalls.blockingUnaryCall(newCall(channel), request);
+  }
+
+  /**
+   * Convenience method to run a unary callable synchronously, without channel. Requires a callable
+   * which is bound to a channel.
+   */
+  public ResponseT call(RequestT request) {
+    return call(null, request);
+  }
+
+  /**
+   * Convenience method to run a unary callable asynchronously. If no channel is provided,
+   * the callable must be bound to one.
+   */
+  public void asyncCall(@Nullable Channel channel, RequestT request,
+      StreamObserver<ResponseT> responseObserver) {
+    ClientCalls.asyncUnaryCall(newCall(channel), request, responseObserver);
+  }
+
+  /**
+   * Convenience method to run a unary callable asynchronously, without channel. Requires a callable
+   * which is bound to a channel.
+   */
+  public void asyncCall(RequestT request, StreamObserver<ResponseT> responseObserver) {
+    asyncCall(null, request, responseObserver);
```

Comment:
Could use the call-type in the descriptor to assert early if someone tried to use the wrong method